core/node/libp2p/rcmgr_defaults.go

```diff
@@ -79,109 +48,79 @@ Run 'ipfs swarm limit all' to see the resulting limits.
 
 		// Keep it simple by not having Service, ServicePeer, Protocol, ProtocolPeer, Conn, or Stream limits.
-		ServiceBaseLimit:     infiniteBaseLimit,
-		ServiceLimitIncrease: noLimitIncrease,
+		ServiceDefault: rcmgr.InfiniteLimits.ToPartialLimitConfig().System,
 
-		ServicePeerBaseLimit:     infiniteBaseLimit,
-		ServicePeerLimitIncrease: noLimitIncrease,
+		ServicePeerDefault: rcmgr.InfiniteLimits.ToPartialLimitConfig().System,
 
-		ProtocolBaseLimit:     infiniteBaseLimit,
-		ProtocolLimitIncrease: noLimitIncrease,
+		ProtocolDefault: rcmgr.InfiniteLimits.ToPartialLimitConfig().System,
 
-		ProtocolPeerBaseLimit:     infiniteBaseLimit,
-		ProtocolPeerLimitIncrease: noLimitIncrease,
+		ProtocolPeerDefault: rcmgr.InfiniteLimits.ToPartialLimitConfig().System,
 
-		ConnBaseLimit:     infiniteBaseLimit,
-		ConnLimitIncrease: noLimitIncrease,
+		Conn: rcmgr.InfiniteLimits.ToPartialLimitConfig().System,
 
-		StreamBaseLimit:     infiniteBaseLimit,
-		StreamLimitIncrease: noLimitIncrease,
+		Stream: rcmgr.InfiniteLimits.ToPartialLimitConfig().System,
 
 		// Limit the resources consumed by a peer.
 		// This doesn't protect us against intentional DoS attacks since an attacker can easily spin up multiple peers.
 		// We specify this limit against unintentional DoS attacks (e.g., a peer has a bug and is sending too much traffic intentionally).
 		// In that case we want to keep that peer's resource consumption contained.
 		// To keep this simple, we only constrain inbound connections and streams.
-		PeerBaseLimit: rcmgr.BaseLimit{
-			Memory:          bigEnough,
-			FD:              bigEnough,
-			Conns:           bigEnough,
-			ConnsInbound:    rcmgr.DefaultLimits.PeerBaseLimit.ConnsInbound,
-			ConnsOutbound:   bigEnough,
-			Streams:         bigEnough,
-			StreamsInbound:  rcmgr.DefaultLimits.PeerBaseLimit.StreamsInbound,
-			StreamsOutbound: bigEnough,
-		},
-		// Most limits don't see an increase because they're already infinite/bigEnough.
-		// The values that should scale based on the amount of memory allocated to libp2p need to increase accordingly.
-		PeerLimitIncrease: rcmgr.BaseLimitIncrease{
-			Memory:          0,
-			FDFraction:      0,
-			Conns:           0,
-			ConnsInbound:    rcmgr.DefaultLimits.PeerLimitIncrease.ConnsInbound,
-			ConnsOutbound:   0,
-			Streams:         0,
-			StreamsInbound:  rcmgr.DefaultLimits.PeerLimitIncrease.StreamsInbound,
-			StreamsOutbound: 0,
+		PeerDefault: rcmgr.ResourceLimits{
+			Memory:          rcmgr.Unlimited64,
+			FD:              rcmgr.Unlimited,
+			Conns:           rcmgr.Unlimited,
+			ConnsInbound:    rcmgr.DefaultLimit,
+			ConnsOutbound:   rcmgr.Unlimited,
+			Streams:         rcmgr.Unlimited,
+			StreamsInbound:  rcmgr.DefaultLimit,
+			StreamsOutbound: rcmgr.Unlimited,
 		},
 	}
 
-	// Whatever limits libp2p has specifically tuned for its protocols/services we'll apply.
-	libp2p.SetDefaultServiceLimits(&scalingLimitConfig)
```

Comment:
This is because we used to build our own `scalingLimitConfig`, we now use the default libp2p scalling config (see `rcmgr.DefaultLimits.Scale` below). This is already included.

Reply:
We discussed this verbally.  This should get added back in so that we see limits get set for scope like protocol.identify.    ``rcmgr.DefaultLimits`` doesn't include the libp2p-owned protocol/service limits.  Those get set with ``libp2p.SetDefaultServiceLimits``.